### PR TITLE
feat: add support for gemma-4-12B-it

### PR DESCRIPTION
**Pull request by MoonRide303**
Use the model's raw unprocessed logits (output_logits) instead of the processed generation scores when computing KL divergence. Fixes KL divergence: nan on google/gemma-4-12B-it.

### CODE REVIEW

**Comment by p-e-w** [comment]
Thanks!

I'm trying to remember why I used scores instead of logits in the first place. I have a vague memory of an incompatibility, but it could be just a hallucination. Otherwise, this solution seems obviously correct.

I do wonder though: What *are* the logits that `generate` returns for those tokens?

**Comment by MoonRide303** [comment]
> I do wonder though: What _are_ the logits that `generate` returns for those tokens?

Just raw model logits (before `SuppressTokensLogitsProcessor`).

**Comment by igorls** [comment]
This is the right fix — I came at #346 from the KL-computation side (#348) but `output_logits` addresses the root cause and keeps `F.kl_div` untouched, which is what the maintainer wanted.

Some verification on `google/gemma-4-12B-it` that should ease any regression concern: for the first greedy token the only active logits processor is `suppress_tokens`, so `outputs.logits[0]` and `outputs.scores[0]` are **identical except at the 2 suppressed token positions** (EOI/EOA). Concretely, on a 2-prompt batch:

```
output_scores: finite=False  num -inf=4   ->  self-KL = nan
output_logits: finite=True   num -inf=0   ->  self-KL = 0.0
elements where scores != logits: 4  (= 2 suppressed tokens x 2 rows)
```

So for every model that doesn't suppress tokens, `output_logits == output_scores` and KL values are unchanged; suppressed models get the correct raw distribution instead of a NaN. `get_residuals` correctly needs no change since it reads `hidden_states`, which processors never touch. I've closed #348 in favor of this.

**Comment by p-e-w** [comment]
Indeed, this is the correct fix. Thank you, merged! And thank you also to @igorls for the valuable information!
